### PR TITLE
feat: [SAM] able to download the result as png

### PR DESCRIPTION
**Pull request by lichin-lin**
@radames I've add a download button and the download function for the cutout image!

https://github.com/huggingface/candle/assets/10290616/5503f55e-623e-402a-9f51-482c9b148a27

### CODE REVIEW

File: candle-wasm-examples/segment-anything/lib-example.html [hunk 1]
```diff
@@ -163,6 +169,45 @@
...
       });
+      // add event to download btn
+      downloadBtn.addEventListener("click", async () => {
+        // Function to load image blobs as Image elements asynchronously
+        const loadImageAsync = (imageURL) => {
+          return new Promise((resolve) => {
+            const img = new Image();
+            img.onload = () => {
+              resolve(img);
+            };
```
Comment (radames) [inline candle-wasm-examples/segment-anything/lib-example.html:180]: can you please add this here? I was getting tainted canvas with the example imagas

```
img.crossOrigin = "anonymous";
```

**Review by radames — CHANGES_REQUESTED** [review]
Very cool! You also need to add `copyImageURL = href;` so the examples can work

https://github.com/huggingface/candle/pull/1035/files#diff-81cb3980e2868df7ea08351b7bceda0764c3a8662e9711d857e45aa510689877R151-R156

**Comment by radames** [comment]
Here is live @LaurentMazare if you want to check! https://huggingface.co/spaces/radames/candle-segment-anything-wasm

**Comment by LaurentMazare** [comment]
That's super cool to play with and easily detour images! One cosmetic thing, rather than "download the result" maybe it could be "download the selected part" or something like that?

**Comment by radames** [comment]
How about "Download Cut-Out" ? 
It's also interesting to play with the model size, you can see the artifacts of the smaller model compared to the larger one.

**Comment by lichin-lin** [comment]
Thanks for the review, will update soon! 🙌
I have a question related to cutout result: why sometime I'll get holes for the cutout mask? (is this something to do with how model generates the embeddings?)


https://github.com/huggingface/candle/assets/10290616/f3856e25-34fb-4510-97ce-9fb82f8333b6

**Comment by radames** [comment]
hi @lichin-lin, as I mentioned on my last message, this is defiantly due the model size MobileSAM the default on our list. if you try SAM base, you can see it'll perform better.

**Comment by lichin-lin** [comment]
Got it, thx for the explanation!
I just pushed a commit to the PR. 🙌

**Comment by LaurentMazare** [comment]
Looks all good to me, @radames any more thoughts or good to merge?

**Review by radames — APPROVED** [review]

**Pull request merged**